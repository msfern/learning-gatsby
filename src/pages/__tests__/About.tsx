import React from "react";
import renderer from "react-test-renderer";
import * as Gatsby from "../../../__mocks__/gatsby";
import About from "../about";

beforeEach(() => {
  const useStaticQuery = jest.spyOn(Gatsby, "useStaticQuery");
  useStaticQuery.mockImplementationOnce(() => ({
    site: {
      siteMetadata: {
        title: "My Title",
      },
    },
  }));
});

describe("About", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<About />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
