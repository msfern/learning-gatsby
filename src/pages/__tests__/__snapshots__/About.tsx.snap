// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About renders correctly 1`] = `
<div
  className="container"
  data-testid="hero-title"
>
  <title>
    About Me
     | 
    My Title
  </title>
  <header
    className="siteTitle"
  >
    My Title
  </header>
  <nav>
    <ul
      className="navLinks"
    >
      <li
        className="navLinkItem"
      >
        <a
          className="navLinkText"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        className="navLinkItem"
      >
        <a
          className="navLinkText"
          href="/about"
        >
          About
        </a>
      </li>
      <li
        className="navLinkItem"
      >
        <a
          className="navLinkText"
          href="/blog"
        >
          Blog
        </a>
      </li>
    </ul>
  </nav>
  <main>
    <h1
      className="heading"
    >
      About Me
    </h1>
    <p>
      Hi there! I'm the proud creator of this site, which I built with Gatsby.
    </p>
  </main>
</div>
`;
